plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystemMinio.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.s3;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.ListObjectsV2Request;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+import com.google.common.collect.AbstractSequentialIterator;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterators;
+import io.minio.MakeBucketArgs;
+import io.minio.MinioClient;
+import io.trino.testing.containers.Minio;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.util.Iterator;
+
+import static io.trino.hadoop.ConfigurationInstantiator.newEmptyConfiguration;
+import static java.util.UUID.randomUUID;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public class TestTrinoS3FileSystemMinio
+        extends AbstractTestTrinoS3FileSystemObjectStorage
+{
+    public static final String MINIO_ACCESS_KEY = "accesskey";
+    public static final String MINIO_SECRET_KEY = "secretkey";
+    private String bucketName = "trino-ci-test";
+
+    private Minio minio;
+
+    private MinioClient minioClient;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        minio =
+                Minio.builder()
+                        .withEnvVars(ImmutableMap.<String, String>builder()
+                                .put("MINIO_ACCESS_KEY", MINIO_ACCESS_KEY)
+                                .put("MINIO_SECRET_KEY", MINIO_SECRET_KEY)
+                                .buildOrThrow())
+                        .build();
+        minio.start();
+
+        minioClient =
+                MinioClient.builder()
+                        .endpoint("http://localhost:" + minio.getMinioApiEndpoint().getPort())
```

Comment:
why ignore minio.getMinioApiEndpoint().getHost()?

btw see [Make it easier to use MinIO with Hadoop](https://github.com/trinodb/trino/pull/14489/commits/f4fa22f0aa2b5ecc3ec4a28aa7310aa34a0d43a6) in https://github.com/trinodb/trino/pull/14489